### PR TITLE
Add CSS variable for Card spacing

### PR DESCRIPTION
**Pull request by fineemb**
## Proposed change
Card spacing cannot be customized, which affects the play of custom themes, such as shadows.
This change provides a CSS variable with adjustable spacing

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @fineemb, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#5738).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/resources/ha-style.ts [hunk 1]
```diff
@@ -92,6 +92,8 @@ documentContainer.innerHTML = `<custom-style>
       --rgb-primary-text-color: 33, 33, 33;
       --rgb-secondary-text-color: 114, 114, 114;
       --rgb-text-primary-color: 255, 255, 255;
+      /* other */
+      --hui-column-margin:4px 4px 8px;
```
Comment (zsarnett) [inline src/resources/ha-style.ts:96]: I don't think we necessarily need to add this to `ha-style`

File: src/panels/lovelace/views/hui-view.ts [hunk 2]
```diff
@@ -148,7 +148,7 @@ export class HUIView extends LitElement {
 
         .column > * {
           display: block;
-          margin: 4px 4px 8px;
+          margin: var(--hui-view-column-margin);
```
Comment (zsarnett) [inline src/panels/lovelace/views/hui-view.ts:151]: referring to the other comment
```suggestion
          margin: var(--hui-view-column-margin, 4px 4px 8px);
```

File: src/resources/ha-style.ts (hunk 1 above)
Comment (zsarnett) [inline src/resources/ha-style.ts:96]:
```suggestion
```

File: src/panels/lovelace/views/hui-view.ts [hunk 3]
```diff
@@ -148,7 +148,7 @@ export class HUIView extends LitElement {
 
         .column > * {
           display: block;
-          margin: 4px 4px 8px;
+          margin: var(--hui-view-column-margin, 4px 4px 8px);
```
Comment (zsarnett) [inline src/panels/lovelace/views/hui-view.ts:151]:
```suggestion
          margin: var(--view-column-margin, 4px 4px 8px);
```

**Review by zsarnett — APPROVED** [review]
Changes look good. Up to Bram if this is something that should be added

**Comment by bramkragten** [comment]
I'm ok with this, if you also find a nice way to handle it with stacks.

**Comment by bramkragten** [comment]
@fineemb ?

**Comment by bramkragten** [comment]
Let's close it then. You can reopen if you found a solution.

**Pull request closed without merging**